include/boost/spirit/home/x3/support/traits/string_traits.hpp

```diff
@@ -93,42 +93,63 @@ namespace boost { namespace spirit { namespace x3 { namespace traits
     template <>
     struct char_type_of<wchar_t> : mpl::identity<wchar_t> {};
 
+    template <>
+    struct char_type_of<char32_t> : mpl::identity<char32_t> {};
```

Comment:
I am inclined towards simplifying the `char_type_of` instead because there are more char types and soon or later the question will rise again.

Reply:
> I am inclined towards simplifying and generalizing the `char_type_of` instead because there are more char types and soon or later the question will rise again.

Makes sense. I'd love to see some ideas in code.

Reply:
See #507

Reply:
Oh cool, that PR obsoletes mine then, no? I'm fine with that. Less code maintenance in general is better.

Reply:
> We can still keep the small tests that I added though as those still have some value.

Absolutely. I just merged #507. Could you please do a final tweak on this PR to deal the changes? Thanks!

Reply:
Alright, took a big of git magic to get there but I managed to condense my PR down to 1 commit that just updates the test files.